templates/hooks/hosted_zone/sdk_create_post_build_request.go.tpl

```diff
@@ -0,0 +1,6 @@
+    // You must use a unique CallerReference string every time you submit a
+    // CreateHostedZone request. CallerReference can be any unique string, for
+    // example, a date/timestamp.
+    // TODO: Name is not sufficient, since a failed request cannot be retried.
+    // We might need to import the `time` package into `sdk.go`
+    input.SetCallerReference(*desired.ko.Spec.Name)
```

Comment:
To reviewers: Please note that this needs to be changed. Looking for suggestions

Reply:
If you wanna avoid importing "time" package. You can use a different file like `hook.go` and create some function like `generateCallerReference`